src/main/java/com/Model/User.java

```diff
@@ -1,57 +1,78 @@
 package com.Model;
 
-import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
 import javax.persistence.*;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
 
 @Entity
-@Inheritance (strategy = InheritanceType.JOINED)
-public class User {
+public class User implements UserDetails {
 
     @Id
     @GeneratedValue(strategy = GenerationType.AUTO)
     private long id;
+
     private String emailAddress;
     private String password; //Encrypted password
```

Comment:
Can't you store the encrypted version instead of the String password? Just encrypt whatever you receive right away and store that instead.

Reply:
I am using BCryptPasswordEncoder. The return type of [encode(CharSequence rawPassword)] is string. 
https://docs.spring.io/spring-security/site/docs/4.2.15.RELEASE/apidocs/org/springframework/security/crypto/bcrypt/BCryptPasswordEncoder.html